### PR TITLE
Create a bootstrap governance document

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Foundation work for #1. The governance document is based on the OpenTelemetry, Keptn and some bits of Contributor Covenant and TAG App Delivery

Preview: https://github.com/openfeatureflags/governance/tree/governance-docs

Signed-off-by: Oleg Nenashev <o.v.nenashev@gmail.com>

### CODE REVIEW

File: README.md
Comment (beeme1mr) [inline README.md:42]: This section deals more with the proposed reference implementation. Perhaps we should omit it for now.

File: README.md
Comment (beeme1mr) [inline README.md:15]:
```suggestion
### Goal
```

File: README.md
Comment (dyladan) [inline README.md:36]: I would quote the whole phrase "rebuilding the wheel"

```suggestion
- Many companies **“rebuilding the wheel”**
```

I would also consider not using smart-quotes to ensure compatibility with the most people

```suggestion
- Many companies **"rebuilding the wheel"**
```

File: README.md
Comment (dyladan) [inline README.md:38]: No need to quote vendor here

```suggestion
- **Vendor lock-in** at code-level
```

File: README.md
Comment (dyladan) [inline README.md:39]:
```suggestion
- **No collaboration** across vendors 
```

File: README.md
Comment (dyladan) [inline README.md:48]: I had to read this line several times to understand what you meant. I would consider rewording it to make it more clear. It is also not clear how this is proposed to be solved

File: README.md
Comment (dyladan) [inline README.md:50]: I can't figure out what this bullet is trying to say

File: README.md
Comment (dyladan) [inline README.md:44]: These bullet points are all negatives of existing solutions. I would propose to rephrase them to make them positives of your proposed solution. For example instead of saying existing solutions have eval logic in SDKs, you could say that openfeature will evaluate all rules on a centralized service. Or instead of saying existing solutions do not have a unified grammar, you could say that you are proposing a unified grammar specification that may be understood by any spec-compliant tooling.

File: README.md
Comment (dyladan) [inline README.md:55]: dead or non-public link

File: README.md
Comment (dyladan) [inline README.md:58]: Roadmap appears to be both a github project (linked above) and a label. It will take additional work to keep these in sync and if they are ever out of sync it will be very confusing. I would consider deciding on one or the other (personally I think a project is the better solution here)

File: README.md
Comment (dyladan) [inline README.md:61]: I would suggest that the pull request on the specification itself constitutes a decision and define the rules for when a pull request can be merged. The wording in this section does not make it clear if a decision made in slack should be automatically approved when it is a pull request.

File: README.md
Comment (dyladan) [inline README.md:69]: I would discourage key discussions from happening in project chat where they are not easily searchable or archived.

File: README.md
Comment (oleg-nenashev) [reply README.md:24]: I thin so, but I will update the wording to make it explicit it is intended state

File: README.md
Comment (dyladan) [inline README.md:63]: It is not clear how to become an interested party or what the responsibilities and privileges are of an interested party. Currently the interested parties are all listed as companies, not individual contributors. Does this mean each of these companies will always have the same power regardless of if their contributors move away from the project? This model also strongly discourages open source contributors who are not affiliated with a company or with a company that wants to make a public declaration of support for the project. Most CNCF projects I'm aware of don't have such a corporate-first model of governing. I would suggest you define roles such as technical committee, governance committee, and maintainer, and make it very clear what the requirements are to move to the next rung of the ladder and what the expectations/permissions/responsibilities are for each. I would also caution against having "founder" be a position at all because you would hope the project lasts longer than any person's involvement or employment at a particular company.

File: README.md
Comment (oleg-nenashev) [reply README.md:42]: I can move it to the specification indeed

**Review by dyladan — COMMENTED** [review]
As a style nit, the punctuation on bulleted lists is not very consistent. I would suggest to just remove all periods at the end of bullet points or add them to all bullet points.

File: README.md
Comment (oleg-nenashev) [reply README.md:11]: Nope, leftover

```suggestion
```

File: README.md
Comment (oleg-nenashev) [reply README.md:43]: This comes from the slidedeck by @beeme1mr. I do not have strong opinion here, because manifesto documents are usually quite opinionated. Happy to rephrase though

File: README.md
Comment (oleg-nenashev) [reply README.md:62]: > It is not clear how to become an interested party or what the responsibilities and privileges are of an interested party. 

The was a sentence about submitting a pull request. Created https://github.com/openfeatureflags/governance/pull/10 to make things more explicit

> Currently the interested parties are all listed as companies, not individual contributors. This model also strongly discourages open source contributors....

I do not think so, the first field was for an individual. Anyway https://github.com/openfeatureflags/governance/pull/10 makes it much more explicit

> Most CNCF projects I'm aware of don't have such a corporate-first model of governing

Well... things in CNCF projects are quite complicated

> I would suggest you define roles such as technical committee, governance committee, and maintainer, and make it very clear what the requirements are to move to the next rung of the ladder and what the expectations/permissions/responsibilities are for each.

This is in the scope for the full governance model. This document is only the **bootstrap** governance document. Created https://github.com/openfeatureflags/governance/issues/11 to be explicit about the expectations for the governance document

> I would also caution against having "founder" be a position at all

There might be a Founder position with some additional privileges in the final governance. I would not support a founder as ultimate decision making body (BDFL alike governance models), and I haver a story of fighting against it in Jenkins.

For the bootstrap governance, having an ultimate decision making body is common and somewhat reasonable to speed up the things. I am waiting feedback/decision by @beeme1mr on this matter

File: README.md
Comment (oleg-nenashev) [reply README.md:68]: Me too. At the same time unfortunately it happens in almost all projects. I am fine with it happening in the bootstrap phase when not all channels are established. I will remove explicit reference in the text anyway

```suggestion
Key discussions and decisions should happen asynchronously in communication channels like GitHub Issues, discussions or pull requests.
```

File: README.md
Comment (oleg-nenashev) [inline README.md:10]:
```suggestion
> is effective until the project is fully established.
> See [this issue](https://github.com/openfeatureflags/governance/issues/11) for the scope of the full governance document.
```

File: README.md
Comment (oleg-nenashev) [reply README.md:23]:
```suggestion
- A **vendor-agnostic ecosystem** for users, application developers, SREs, and vendors
```

File: README.md
Comment (oleg-nenashev) [inline README.md:16]:
```suggestion
Our goal is to create an open standard for feature flag management to support a robust feature flag ecosystem using cloud native technologies.
```

File: README.md
Comment (oleg-nenashev) [inline README.md:26]:
```suggestion
- Extensibility for open source and commercial offerings
- Adoption by the open source and proprietary projects listed on the [Cloud Native Landscape](https://landscape.cncf.io/)
```

File: README.md
Comment (oleg-nenashev) [inline README.md:40]:
```suggestion

## CNCF membership

Our goal is to join the [Cloud Native Computing Foundation (CNCF)](https://www.cncf.io/) as a sandbox project shortly after the inception of OpenFeature.
Should it be accepted, the project will establish governance along with the CNCF best practices, 
and then target reaching the incubating and than the graduated statuses in the foundation.

```

File: README.md
Comment (oleg-nenashev) [reply README.md:61]: FTR https://github.com/openfeatureflags/governance/issues/11

File: README.md
Comment (dyladan) [reply README.md:59]: I guess it depends what you mean by roadmap. If you mean it to refer to overall project direction, then I would argue that only certain people in technical leadership should have permission to update it anyway. If you mean it more like a backlog that is just any task planned for the future then a label would be fine. In any case, you can do what you feel is best here its just my opinion.

File: README.md
Comment (oleg-nenashev) [inline README.md:41]:
```suggestion
### Why OpenFeature?

There are many existing solutions, open source or not.
In the industry there are multiple common issues limiting wide adoption:

- **Many SDKs** doing the same thing, but slightly differently
- SDKs are generally open source but have no ecosystem
- **Different APIs** in implementations, hence it is difficult to migrate between them
- **Limited interoperability** across implentations 
- **Missing integration** with observability tools (e.g. OpenTelemetry)
```

File: README.md
Comment (oleg-nenashev) [inline README.md:52]:
```suggestion
* Current solutions often have **eval logic in SDKs**.
  This may require **updating all your apps** for a new rule type.
* Config in Feature Flagging tools that are **separate from the rest of the app** make it harder to use approaches like GitOps. 
* Config and rule changes are **hard to integrate into observability** tools 
* Information to decide on features already available in observability tools is **not leveraged and collected twice**
* No standardized format for defining feature flagging rules
```

File: README.md
Comment (dyladan) [reply README.md:42]: Seems fine but as I said above I would suggest to define the governance model as early as possible in order to grow a healthy community. It will be easier to convince people to join the project in early days if they see a clear path for them to become an important part of the project (maintainer, governance, technical committee, etc).

File: README.md
Comment (beeme1mr) [inline README.md:3]:
```suggestion
[![Roadmap](https://img.shields.io/static/v1?label=Roadmap&message=public&color=green)](https://github.com/orgs/open-feature/projects/1)
```

File: README.md
Comment (beeme1mr) [inline README.md:4]:
```suggestion
[![Contributing](https://img.shields.io/static/v1?label=Contributing&message=guide&color=blue)](https://github.com/open-feature/.github/blob/main/CONTRIBUTING.md)
```

File: README.md
Comment (beeme1mr) [inline README.md:38]:
```suggestion
- **Limited interoperability** across implementations
```

File: README.md
Comment (beeme1mr) [inline README.md:11]:
```suggestion
> See [this issue](https://github.com/open-feature/governance/issues/11) for the scope of the full governance document.
```

**Comment by oleg-nenashev** [comment]
FTR I approve the changes and not blocking merge. I would prefer to have separate governance and community repositories, but we can tackle it later. Will create a follow-up GitHub issue

**Review by dyladan — APPROVED** [review]

**Pull request merged**
